src/v3/src/components/TextWithActionLink/TextWithActionLink.tsx

```diff
@@ -31,45 +31,35 @@ const TextWithHtml: UISchemaElementComponent<{
     actionParams,
     step,
     stepToRender,
-    submitOnClick,
     contentClassname,
     isActionStep,
   } = uischema.options;
   const onSubmitHandler = useOnSubmit();
+  const parsedContent = useHtmlContentParser(content);
 
   const handleClick = async (e: Event) => {
-    if (submitOnClick) {
-      e.preventDefault();
+    e.preventDefault();
 
-      if (loading) {
-        return;
-      }
+    if (loading) {
+      return;
+    }
 
-      // only submit when className matches
-      if ((e.target as HTMLElement).className.includes(contentClassname)) {
-        onSubmitHandler({
-          step,
-          stepToRender,
-          params: actionParams,
-          isActionStep,
-        });
-      }
+    // only submit when className matches
+    if ((e.target as HTMLElement).className.includes(contentClassname)) {
+      onSubmitHandler({
+        step,
+        stepToRender,
+        params: actionParams,
+        isActionStep,
+      });
     }
   };
 
   return (
-    <Box>
-      <Box
-        sx={(theme) => ({
-          [`& .${contentClassname}`]: {
-            color: theme.palette.primary.main,
-          },
-        })}
```

Comment:
the parser automatically converts anchor tags to the ODS MUI link component, so no need to override the color.